server/lib/scene/scene.addScene.js

```diff
@@ -31,65 +31,68 @@ function addScene(sceneRaw) {
   const scene = cloneDeep(sceneRaw);
   // first, if the scene actually exist, we cancel all triggers
   this.cancelTriggers(scene.selector);
-  // Foreach triggger, we schedule jobs for triggers that need to be scheduled
-  if (scene.triggers) {
-    scene.triggers.forEach((trigger) => {
-      // First, we had a trigger key, import to uniquely identify this trigger
-      trigger.key = uuid.v4();
-      if (trigger.type === EVENTS.TIME.CHANGED && trigger.scheduler_type !== 'interval') {
-        const rule = new schedule.RecurrenceRule();
-        rule.tz = this.timezone;
-        switch (trigger.scheduler_type) {
-          case 'every-month':
-            rule.date = trigger.day_of_the_month;
-            rule.hour = parseInt(trigger.time.substr(0, 2), 10);
-            rule.minute = parseInt(trigger.time.substr(3, 2), 10);
-            rule.second = 0;
-            break;
-          case 'every-week':
-            rule.dayOfWeek = trigger.days_of_the_week.map((day) => nodeScheduleDaysOfWeek[day]);
-            rule.hour = parseInt(trigger.time.substr(0, 2), 10);
-            rule.minute = parseInt(trigger.time.substr(3, 2), 10);
-            rule.second = 0;
-            break;
-          case 'every-day':
-            rule.hour = parseInt(trigger.time.substr(0, 2), 10);
-            rule.minute = parseInt(trigger.time.substr(3, 2), 10);
-            rule.second = 0;
-            break;
-          case 'custom-time':
-            rule.year = parseInt(trigger.date.substr(0, 4), 10);
-            rule.month = parseInt(trigger.date.substr(5, 2), 10) - 1;
-            rule.date = parseInt(trigger.date.substr(8, 4), 10);
-            rule.hour = parseInt(trigger.time.substr(0, 2), 10);
-            rule.minute = parseInt(trigger.time.substr(3, 2), 10);
-            rule.second = 0;
-            break;
-          default:
-            throw new BadParameters(`${trigger.scheduler_type} not supported`);
-        }
-        trigger.nodeScheduleJob = schedule.scheduleJob(rule, () => this.event.emit(EVENTS.TRIGGERS.CHECK, trigger));
-      } else if (trigger.type === EVENTS.TIME.CHANGED && trigger.scheduler_type === 'interval') {
-        let intervalMilliseconds;
-        switch (trigger.unit) {
-          case 'second':
-            intervalMilliseconds = trigger.interval * 1000;
-            break;
-          case 'minute':
-            intervalMilliseconds = trigger.interval * 60 * 1000;
-            break;
-          case 'hour':
-            intervalMilliseconds = trigger.interval * 60 * 60 * 1000;
-            break;
-          default:
-            throw new BadParameters(`${trigger.unit} not supported`);
-        }
-        if (intervalMilliseconds > MAX_VALUE_SET_INTERVAL) {
-          throw new BadParameters(`${trigger.interval} ${trigger.unit} is too big for an interval`);
+
+  if (scene.active) {
+    // Foreach triggger, we schedule jobs for triggers that need to be scheduled
+    if (scene.triggers) {
```

Comment:
Why not both if in single one?

If scene is active and have trigger, no.

Reply:
You are right (Y)